app/proc/telegram.go

```diff
@@ -45,18 +47,38 @@ func (client TelegramClient) Send(channelID string, item feed.Item) error {
 		return nil
 	}
 
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	response := make(chan *responseTelegram)
+	go client.send(channelID, item, response)
+
+	select {
```

Comment:
it looks wrong as it leaks goroutine on timeout. Timeout support has to be or on telegram's client level or on the level of http client injected to the telegram client.

Reply:
see https://github.com/tucnak/telebot/blob/v2/bot.go#L69 - you can pass http client as set timeout for the client, without any need in this dangerous workaround with goroutine.

Reply:
use http.Client https://github.com/umputun/feed-master/commit/69f08f9d8eff5cff634973b12eeea0b48f28c53c